### PR TITLE
Fix chinese slugify

### PR DESCRIPTION
**Pull request by Joey777210**
**What is this feature?**
Fix Chinese character lose when slugify dashboard titles.

[Add a brief description of what the feature or update does.]

**Why do we need this feature?**
Anyone use grafana use Chinese.

[Add a description of the problem the feature is trying to solve.]

**Who is this feature for?**
Anyone use grafana use Chinese.

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:
#66373 

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ √] It works as expected from a user's perspective.
- [ √] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -1919,6 +1919,8 @@ github.com/morikuni/aec v1.0.0/go.mod h1:BbKIizmSmc5MMPqRYbxO4ZU0S0+P200+tUnFx7P
 github.com/moul/http2curl v0.0.0-20170919181001-9ac6cf4d929b/go.mod h1:8UbvGypXm98wA/IqH45anm5Y2Z6ep6O31QGOAZ3H0fQ=
 github.com/moul/http2curl v1.0.0/go.mod h1:8UbvGypXm98wA/IqH45anm5Y2Z6ep6O31QGOAZ3H0fQ=
 github.com/mozilla/tls-observatory v0.0.0-20190404164649-a3c1b6cfecfd/go.mod h1:SrKMQvPiws7F7iqYp8/TX+IhxCYhzr6N/1yb8cwHsGk=
+github.com/mozillazg/go-pinyin v0.20.0 h1:BtR3DsxpApHfKReaPO1fCqF4pThRwH9uwvXzm+GnMFQ=
```
Comment (ryantxu) [inline go.sum:1922]: how big are these libraries?  we supported this before, but the library was 5mb 😬

**Comment by sakjur** [comment]
👋 I'm thinking about an alternative where instead of pinyin (and similar solutions for other alphabets) we might want to put some unicode codepoint based encoding in the slug, since that'd be generally applicable.

**Comment by Joey777210** [comment]
Sure, that would be better, as long as the result of slug is readable.

**Comment by Joey777210** [comment]
> 👋 I'm thinking about an alternative where instead of pinyin (and similar solutions for other alphabets) we might want to put some unicode codepoint based encoding in the slug, since that'd be generally applicable.

Have you found any solution？ I haven't found one yet.

**Comment by alexandnpu** [comment]
> 👋 I'm thinking about an alternative where instead of pinyin (and similar solutions for other alphabets) we might want to put some unicode codepoint based encoding in the slug, since that'd be generally applicable.

I am wondering why the original lib was removed, just because of its size? For me, the 5MB size is reasonable.  There are still many alphabets to support.

I agree that we have to find a solution to support not only Chinese but also other langugages.

**Comment by sakjur** [comment]
> I am wondering why the original lib was removed, just because of its size? For me, the 5MB size is reasonable. There are still many alphabets to support.

Forgot to respond to this.
Yes, I believe that that was it. We've gradually removed the prominence of slugs since the introduction of folders in Grafana 5 to replace with titles and UIDs (that can be user defined). Eventually, it's possible that even having a slug library is going to be unnecessary.

It's not wholly unreasonable to add 5 MB to the download size of Grafana for slugs to work well, the bigger problem here was that once it broke, fixing that would risk breaking new entries, so simply reintroducing the library wouldn't have solved that problem.

In #70691, we're using urlencoding to actually encode the utf8 characters into the url for sufficiently short titles, which is a generalized alternative to this that adds almost no weight to the final binary. Closing this in favor of that.

**Pull request closed without merging**